sources/Application/Instruments/MidiInstrument.cpp

```diff
@@ -63,18 +63,25 @@ bool MidiInstrument::Start(int c, unsigned char note, bool retrigger) {
   velocity_ = msg.data2_;
   playing_ = true;
   retrig_ = false;
+  porto_ = false;
 
   return true;
 };
 
 void MidiInstrument::Stop(int c) {
+  char note = lastNote_[c];
+  if (porto_) {
+    porto_ = false;
+    porto_pending_[c] = note;
+    return;
```

Comment:
I don't really understand the flow here. Once you return here, what triggers a second Stop in order to evaluate the function again and execute lines 112+?

Reply:
The idea I here is that if a "stop note" event comes in while there is a prev note sounding and the porto command is in effect, then we don't send a stop midi command here, instead we save it in `porto_pending_[c]` so that the code in Render will send it instead after it sends the next "start note" midi command, so that we only send a stop note command after sending a start note. 
I admit it is a rather convoluted, I'm open for suggestions on any way to make this code cleaner and clearer to understand.